File: Arrays/Move_zero_to_end/brute_sol.py
def moveZeroes(arr):

    n = len(arr)
    temp = []
    for i in range(n):
        if arr[i] != 0:
            temp.append(arr[i])

    nz = len(temp)
    for i in range(nz):
        arr[i] = temp[i]

    for i in range(nz,n):
        arr[i] = 0

    return arr


if __name__ == "__main__":
    arr = [1,2,0,0,4,0,6]
    print(moveZeroes(arr))

File: Arrays/Move_zero_to_end/brute_sol.cpp
#include<bits/stdc++.h>
using namespace std;

vector<int> moveZeroes(int n, vector<int> a){

    //Step-1
    vector<int> temp;
    for(int i=0; i<n; i++){
        if (a[i] != 0){
            temp.push_back(a[i]);
        }
    }

    //Step-2
    int nz = temp.size();
    for(int i=0; i<nz; i++){
        a[i] = temp[i];
    }

    //Step-3
    for(int i=nz; i<n; i++){
        a[i] = 0;
    }

    return a;

}


int main(){
    vector<int> arr = {0, 1, 0, 3, 12};
    int n = arr.size();
    vector<int> result = moveZeroes(n,arr);

    // Print result
    cout << "Array after moving zeroes: ";
    for (int num : result) {
        cout << num << " ";
    }
    cout << endl;
    return 0;
}